### ISSUE TITLE
Importing cell reg from matlab and port it to Python

### ISSUE THREAD

**Issue by gidili** [MetaCell/minian#55]
Link --> https://github.com/zivlab/CellReg

**Comment by kneuron** [comment]
from conversation with Will on slack:

idea is that zivlab/CellReg would be used to replace functionality only in the cross_registration.ipynb

should stay as a separate notebook from 'pipeline', to be optionally run afterward to register/associate cells analyzed in multiple sessions (it would only work after you've run 2+ sessions)

(side thought question on that - does the system expect the same parameters to have been used to analyze the two different sessions? I suppose some you might want to make sure are the same, and some you might need to tweak re: specific recording conditions on different days)

**Comment by zsinnema** [comment]
I've installed Matlab on my linux laptop and ran the CellReg GUI, this works fine

At this moment I see 2 options for integration:
1. Running Matlab and use the CellReg GUI
2. Use the Matlab Python engine for running parts of the CellReg module

Note: for both options the enduser (and we) will need to have a Matlab installation on the computer.

Ad 1.
In case it's no problem to have a non Minian UI then we could opt for this option.
Impact on Minian: convert the Minian dataset structure to the CellReg (file) dataset structure

Ad 2.
In case we would like to integrate (parts of) the CellReg module we have to investigate the "API" of the CellReg module. There is no documentation in the project. We could contact the developer(s) to ask if they can help us with doing that. 
There is a Matlab python engine available see https://nl.mathworks.com/help/matlab/matlab-engine-for-python.html

**Comment by gidili** [comment]
@zsinnema is options 2 a way to invoke that code from directly Python
without porting into Python?

**Comment by zsinnema** [comment]
@gidili correct, we can call Matlab functions within the python code. This only works if the (end)user has Matlab and the Matlab python engine installed

**Comment by gidili** [comment]
@zsinnema yeah that sounds like it could 1 save some time 2) ensure we are
not introducing issues, so if the constraint of having matlab installed is
acceptable this could be a good option
